utils/_context/_scenarios/parametric.py

```diff
@@ -354,15 +354,15 @@ def python_library_factory() -> APMLibraryTestServer:
         container_img="""
 FROM ghcr.io/datadog/dd-trace-py/testrunner:9e3bd1fb9e42a4aa143cae661547517c7fbd8924
 WORKDIR /app
-RUN pyenv global 3.9.16
-RUN python3.9 -m pip install fastapi==0.89.1 uvicorn==0.20.0
+RUN pyenv global 3.11
```

Comment:
nit: Do we not want to specify a patch version like we did with 3.9.16?

Reply:
nit: Also what is stopping us from going directly to 3.12 in this PR? Is it just a matter of making our way to 3.12 incrementally?